qualcoder/add_item_name.py

```diff
@@ -93,6 +94,75 @@ def get_new_name(self):
 
         return self.newItem
 
+class DialogLinkTo(QtWidgets.QDialog):
+    """
+    Dialog to get a new code or code category from user.
+    Also used for Case and File adding attributes.
+    Requires a name for Dialog title (and label in setupUI)
+    Requires a list of dictionary 'name' items.
+    Dialog returns ok if the item is not a duplicate of a name in the list.
+    Returns one item through getnewItem method.
+    """
+
+    def __init__(self, model, linktypes, fromname, parent=None):
```

Comment:
I am not sure of what you have done here, hope it works.
The other thing is that you have setup all the widgets here. I prefer to use the qtdesigner UI file for the dialog widget setup. as it is easier to make changes later on.